### PR TITLE
#820: create simple regression test suite

### LINKED ISSUE #820: Create simple regression test suite
Currently, validating the deployment of a new release on TEST or PROD stages
requires a human to check if everything 'still works'.
This process can be supported by implementing a regression test suite which contains test cases for various combinations of topics, indicators, and areas of interest.

Such a test suite is then easily executed as HTTP requests with a simple command line call after the deployment.
(We might even include this in the Jenkins build in the long run, if we switch to continuous integration for TEST.)

Suggested tool:
https://hurl.dev

* command line tool
* simple text syntax
* PyCharm plugin
* uses curl internally

### PR DESCRIPTION
**Pull request by mmerdes**
### Description

This PR adds a regression test suite based on `hurl`.

### Corresponding issue

Closes #820 

### New or changed dependencies

- In order to run the tests, the `hurl` CLI tool must be installed.
- no new Python dependencies 

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
-  ~My code follows the [style guide](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#pre-commit) before committing~
- [x] I have commented my code
- ~[ ] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-api/blob/main/docs/development_setup.md#tests)~
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-api/blob/main/CHANGELOG.md)

Please check all finished tasks. If some tasks do not apply to your PR, please cross their text out (by using `~...~`) and remove their checkboxes.

### CODE REVIEW

File: regression-tests/__run_hurl_tests_for_stage.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+rm -rf report
+mkdir report
+
+hurl *.hurl   --report-html report
+
+echo "\n\nhurl report: file://$PWD/report/index.html"
+
+# mac only:
+# open report/index.html
```
Comment (joker234) [inline regression-tests/__run_hurl_tests_for_stage.sh:11]:
```suggestion
printf "\n\nhurl report: file://$PWD/report/index.html\n"

if [[ "$OSTYPE" == "linux-gnu"* ]]; then
  xdg-open report/index.html
elif [[ "$OSTYPE" == "darwin"* ]]; then
  open report/index.html
else
  printf "\nOS could not be detected. Please open report manually!\n"
fi
```

File: regression-tests/__run_hurl_tests_for_stage.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# requires the env variable $HURL_base_url
```
Comment (joker234) [inline regression-tests/__run_hurl_tests_for_stage.sh:1]:
```suggestion
#!/bin/bash

# requires the env variable $HURL_BASE_URL
```

File: regression-tests/run_hurl_tests_DEV.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+export HURL_base_url=http://127.0.0.1:8080
+
+./__run_hurl_tests_for_stage.sh
```
Comment (joker234) [inline regression-tests/run_hurl_tests_DEV.sh:4]:
```suggestion
cd "$(dirname "$0")"
./__run_hurl_tests_for_stage.sh
```

File: regression-tests/run_hurl_tests_PROD.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+export HURL_base_url=https://api.quality.ohsome.org/v1
+
+./__run_hurl_tests_for_stage.sh
```
Comment (joker234) [inline regression-tests/run_hurl_tests_PROD.sh:4]:
```suggestion
cd "$(dirname "$0")"
./__run_hurl_tests_for_stage.sh
```

File: regression-tests/run_hurl_tests_TEST.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+export HURL_base_url=https://api.quality.ohsome.org/v1-test
+
+./__run_hurl_tests_for_stage.sh
```
Comment (joker234) [inline regression-tests/run_hurl_tests_TEST.sh:4]:
```suggestion
cd "$(dirname "$0")"
./__run_hurl_tests_for_stage.sh
```

**Review by joker234 — COMMENTED** [review]
thanks for the PR!

some remarks:
* env variables should always be upper case: https://bertvv.github.io/cheat-sheets/Bash.html#variables
* I suggest to change the directory to the directory the tests are, otherwise you delete the folders somewhere else or the scripts are not executing properly
* I suggested a OS detection for the report opening in the end
* always state which shell should execute the script to avoid shell specific errors on systems with different default shells

File: regression-tests/buildingarea_adminarea_buildingcomparison.hurl [hunk 6]
```diff
@@ -0,0 +1,23 @@
+POST {{base_url}}/indicators/building-comparison
```
[the remark this answers is not in the document]
Comment (mmerdes) [reply regression-tests/buildingarea_adminarea_buildingcomparison.hurl:1]: The 'mixed-case' spelling was inspired by the hurl convention (or so it seems) to use lower case variables within in the script. But I am fine with giving the bash convention higher priority.

File: regression-tests/__run_hurl_tests_for_stage.sh (hunk 1 above)
Comment (mmerdes) [reply regression-tests/__run_hurl_tests_for_stage.sh:11]: fancy stuff! 🥁 - I trust your scripting more than mine ;)

**Comment by mmerdes** [comment]
@joker234  
Thanks for the review comments - good to have a scripting expert around ;)

**Review by joker234 — APPROVED** [review]

**Pull request merged**